pulser-core/pulser/sequence/sequence.py

```diff
@@ -469,22 +478,20 @@ def _set_slm_mask_dmm(self, dmm_id: str, targets: set[QubitId]) -> None:
             if dmm_id in key:
                 self._slm_mask_dmm = key
                 break
-        # If a channel already contains a slot of type Pulse
-        # Add it to the slm mask DMM just declared.
-        for ch_name, ch_schedule in self._schedule.items():
-            if (
-                isinstance(ch_schedule, _DMMSchedule)
-                or not ch_schedule._has_pulse_slot
-            ):
-                continue
-            for slot in ch_schedule.slots[1:]:
-                if isinstance(
-                    slot.type, Pulse
-                ) and not ch_schedule.is_detuned_delay(slot.type):
-                    self._add_slm_pulse(slot.type, ch_name)
-                    break
-            if self._schedule[cast(str, self._slm_mask_dmm)]._has_pulse_slot:
-                break
+        # Add an SLM pulse if pulses have already been added to Global Channels
+        slm_mask_times = self._schedule.find_slm_mask_times()
+        if slm_mask_times:
+            max_amp = max(
+                [
+                    np.max(ch_schedule.get_samples().amp[: slm_mask_times[1]])
+                    for ch_schedule in self._schedule.values()
+                    if not isinstance(ch_schedule, _DMMSchedule)
+                    and ch_schedule.channel_obj.addressing == "Global"
+                ]
+            )
```

Comment:
Why not just go get the first pulse instead? Sampling is a bit expensive and doesn't seem necessary here

Reply:
I want to be sure that all the amplitudes are way smaller than the detuning of the DMM used for the SLM during the SLM mask time, that's why I found it necessary here. If I was to go only for the first pulse, I don't think I would use `find_slm_mask_times`